pkg/deploy/strategy/recreate/recreate.go

```diff
@@ -121,27 +104,15 @@ func (s *RecreateDeploymentStrategy) Deploy(deployment *kapi.ReplicationControll
 		}
 	}
 
-	// Execute any post-hook.
-	if deploymentConfig.Template.Strategy.RecreateParams != nil {
-		postHook := deploymentConfig.Template.Strategy.RecreateParams.Post
-		if postHook != nil {
-		postHookLoop:
-			for {
-				err := s.hookExecutor.Execute(postHook, deployment)
-				if err == nil {
-					glog.Info("Post hook finished successfully")
-					break
-				}
-				switch postHook.FailurePolicy {
-				case deployapi.LifecycleHookFailurePolicyIgnore, deployapi.LifecycleHookFailurePolicyAbort:
-					// Abort isn't supported here, so treat it like ignore.
-					glog.Infof("Post hook failed, ignoring: %s", err)
-					break postHookLoop
-				case deployapi.LifecycleHookFailurePolicyRetry:
-					glog.Infof("Post hook failed, retrying: %s", err)
-					time.Sleep(s.retryPeriod)
-				}
-			}
+	// Execute any post-hook. Errors are logged and ignored.
+	if params != nil && params.Post != nil {
+		// TODO: handle this in defaulting/conversion/validation?
+		if params.Post.FailurePolicy == deployapi.LifecycleHookFailurePolicyAbort {
+			params.Post.FailurePolicy = deployapi.LifecycleHookFailurePolicyIgnore
```

Comment:
We are completely suppressing failures here. In the very least, we need to create an event or log it. The event/log should mention that the failure policy was Abort, but we couldn't do much about it.